src/mylinef.cpp

```diff
@@ -313,45 +318,40 @@ MyLineF::intersects_flsiOrigX(const QLineF &l, QPointF *intersectionPoint) const
 
 		// Colinear segments
 		const QPointF p2 = l.p2() - p1();
-		qreal n1 = -std::fma(a.x(), c.x(), a.y() * c.y());
+		const qreal n1 = -std::fma(a.x(), c.x(), a.y() * c.y());
 		const qreal n2 = std::fma(a.x(), p2.x(), a.y() * p2.y());
 
 		if (std::signbit(n1) != std::signbit(n2) || n1 < 0) {
-			n1 = std::max(n1, n2);
-			if (n1 > length) {
+			const qreal n = std::max(n1, n2);
+			if (n > length) {
 				if (intersectionPoint)
 					*intersectionPoint = pointAt(0.5 * length);
 				return BoundedIntersection;
 			}
 			else  {
 				if (intersectionPoint)
-					*intersectionPoint = pointAt(0.5 * n1);
-				return n1 < 0 ? UnboundedIntersection : BoundedIntersection;
+					*intersectionPoint = pointAt(0.5 * n);
+				return n < 0 ? UnboundedIntersection : BoundedIntersection;
 			}
 		}
 		else  {
-			n1 = std::min(n1, n2);
-			if (n1 > length)  {
+			const qreal n = std::min(n1, n2);
+			if (n > length)  {
 				if (intersectionPoint)
-					*intersectionPoint = pointAt(0.5 * (length + n1));
+					*intersectionPoint = pointAt(0.5 * (length + n));
 				return UnboundedIntersection;
 			}
 			else  {
 				if (intersectionPoint)
-					*intersectionPoint = (n2 > length) ? pointAt(0.5 * (length + n1)) : pointAt(0.5 * (n1 + n2));
+					*intersectionPoint = (n2 > length) ? pointAt(0.5 * (length + n)) : pointAt(0.5 * (n1 + n2));
 				return BoundedIntersection;
 			}
 		}
 	}
 
-	const qreal na = std::fma(b.y(), c.x(), -b.x() * c.y());
-	if (intersectionPoint)  {
-		const qreal scale = 1 / denominator;		// Do the division iff we really need to
-		*intersectionPoint = QPointF(
-			scale * std::fma(a.x(), na, denominator * p1().x()),
-			scale * std::fma(a.y(), na, denominator * p1().y())
-		);
-	}
+	const qreal na = b.y() * c.x() - b.x() * c.y();
+	if (intersectionPoint)	// Do the division iff we really need to
+		*intersectionPoint = p1() + a * (na / denominator);
```

Comment:
We could improve here by taking:
`qreal scale = 1 / denominator;`
and doing the small parameter expansion:
```
const qreal c = std::fma(a.x(), b.y(), -axby);
na = scale * (na + (na * scale * c));
```
before solving for the intersection point, which'd squeeze a couple of epsilons of precision, but I doubt it's necessary.